### ISSUE TITLE
Refactor the crate into a Cargo workspace.

### ISSUE THREAD

**Issue by filippodebortoli** [phillord/horned-owl#53]
First mentioned [here](https://github.com/phillord/horned-owl/pull/51). I open an issue to make sure this is not forgotten.

The idea is to split the `horned-owl` library from the `horned` tool suite, obtaining a library crate and a binary crate as a result.

One idea could be to use Cargo [workspaces](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) and reorganize the repository.

**Comment by phillord** [comment]
I have made an attempt at this now.

https://github.com/phillord/horned-owl/tree/feature/multi-crate-2

I also tried the other layout with "horned-owl" and "horned-bin" at top level. Better in some ways since `cargo test` will run all workspaces in that way. But I have to move a very large number of files to achieve this, which might make future merges harder. Overall I think this works better. It fits the model that "horned-owl" is the "main" crate".

Comments welcome. I plan to merge this and make a new release as I think this is a big enough change by itself.
